ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryIdConverterFactory.java

```diff
@@ -175,7 +176,11 @@ private CompletionStage<String> convertId(String input, RestRequest restRequest,
             Utils.addSecondsToEpochTime(properties.getCreationTimeInMs(), properties.getTimeToLiveInSeconds());
         NamedBlobRecord record = new NamedBlobRecord(namedBlobPath.getAccountName(), namedBlobPath.getContainerName(),
             namedBlobPath.getBlobName(), blobId, expirationTimeMs);
-        conversionFuture = getNamedBlobDb().put(record).thenApply(result -> result.getInsertedRecord().getBlobId());
+        NamedBlobState state = NamedBlobState.READY;
+        if (properties.getTimeToLiveInSeconds() == Utils.Infinite_Time) {
+          state = NamedBlobState.IN_PROGRESS;
```

Comment:
so when does the record go from in_progress to ready ? after the ttl-update ?

Reply:
In the ttl-update success callback, added comment about it in latest commit.